includes/REST_API/Products_Controller.php

```diff
@@ -46,193 +46,200 @@
-		register_rest_route(
-			$this->namespace,
-			'/' . $this->rest_base,
-			[
-				[
-					'methods'             => WP_REST_Server::READABLE,
-					'callback'            => [ $this, 'get_items' ],
-					'permission_callback' => [ $this, 'get_items_permissions_check' ],
-					'args'                => $this->get_collection_params(),
-				],
-				'schema' => [ $this, 'get_public_item_schema' ],
-			]
-		);
-	}
-
-	/**
-	 * Checks if a given request has access to get and create items.
-	 *
-	 * @since 1.20.0
-	 *
-	 * @param WP_REST_Request $request Full details about the request.
-	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
-	 */
-	public function get_items_permissions_check( $request ) {
-		if ( ! $this->story_post_type->has_cap( 'edit_posts' ) ) {
-			return new \WP_Error(
-				'rest_forbidden',
-				__( 'Sorry, you are not allowed to manage publisher logos.', 'web-stories' ),
-				[ 'status' => rest_authorization_required_code() ]
-			);
-		}
+class Products_Controller extends REST_Controller implements HasRequirements
+{
 
-		return true;
-	}
-
-	/**
-	 * Retrieves all products.
-	 *
-	 * @since 1.20.0
-	 *
-	 * @param WP_REST_Request $request Full details about the request.
-	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
-	 */
-	public function get_items( $request ) {
-		$results = [];
-
-		// TODO(#11154): Refactor to extract product query logic out of this controller.
-		$shopify_host         = $this->settings->get_setting( Settings::SETTING_NAME_SHOPIFY_HOST );
-		$shopify_access_token = $this->settings->get_setting( Settings::SETTING_NAME_SHOPIFY_ACCESS_TOKEN );
-
-		if ( ! empty( $shopify_host ) && ! empty( $shopify_access_token ) ) {
-			return $this->get_items_shopify( $request );
-		}
+    /**
+     * Settings instance.
+     *
+     * @var Settings Settings instance.
+     */
+    private $settings;
+
+    /**
+     * Story_Post_Type instance.
+     *
+     * @var Story_Post_Type Story_Post_Type instance.
+     */
+    private $story_post_type;
+
+    /**
+     * Constructor.
+     *
+     * @param Settings        $settings Settings instance.
+     * @param Story_Post_Type $story_post_type Story_Post_Type instance.
+     */
+    public function __construct(Settings $settings, Story_Post_Type $story_post_type)
+    {
+        $this->settings        = $settings;
+        $this->story_post_type = $story_post_type;
+
+        $this->namespace = 'web-stories/v1';
+        $this->rest_base = 'products';
+    }
 
-		if ( function_exists( 'wc_get_products' ) ) {
-			return $this->get_items_woocommerce( $request );
-		}
+    /**
+     * Get the list of service IDs required for this service to be registered.
+     *
+     * Needed because the story post type needs to be registered first.
+     *
+     * @since 1.20.0
+     *
+     * @return string[] List of required services.
+     */
+    public static function get_requirements(): array
+    {
```

Comment:
Odd that our PHPCS config does not catch this, but we don't usually put the opening `{` bracket on the next line in our code base. Somehow in this file _all brackets_ are now suddenly on the next line.

If you could fix that, that would be great.

Reply:
That file is fixed up now.